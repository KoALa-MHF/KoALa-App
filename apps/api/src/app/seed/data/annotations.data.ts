import { Annotation } from '../../annotations/entities/annotation.entity';

export const AnnotationsData: Partial<Annotation>[] = [
  {
    id: 1,
    markerId: 1,
    userSessionId: 1,
    start: 200,
  },
];
