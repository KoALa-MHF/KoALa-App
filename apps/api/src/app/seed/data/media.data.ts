import { Media } from '../../media/entities/media.entity';

export const MediaData = [
  {
    id: 1,
    name: 'test.mp3',
    mimeType: 'audio/mpeg',
  },
] as [Media];
