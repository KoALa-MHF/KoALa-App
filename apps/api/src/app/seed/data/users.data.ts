export const UsersData = {
  sessionOwner1: {
    id: 1,
    displayName: 'Session Owner 1',
    email: 'session-owner-1@koala-app.de',
    samlId: '123456',
  },
  sessionOwner2: {
    id: 2,
    displayName: 'Session Owner 2',
    email: 'session-owner-2@koala-app.de',
    samlId: '654321',
  },
  sessionParticipant1: {
    id: 3,
    displayName: 'Session Participant 1',
    email: 'session-participant-1@koala-app.de',
  },
  sessionParticipant2: {
    id: 4,
    displayName: 'Session Participant 2',
    email: 'session-participant-2@koala-app.de',
  },
};
