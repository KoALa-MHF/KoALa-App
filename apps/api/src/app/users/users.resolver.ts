import { Resolver, Mutation, Args, Int, Query } from '@nestjs/graphql';
import { UsersService } from './users.service';
import { User } from './entities/user.entity';
import { UpdateUserInput } from './dto/update-user.input';
import { UseGuards } from '@nestjs/common';
import { AuthGuard } from '../core/guards/auth.guard';
import { CurrentUser } from '../core/decorators/user.decorator';

@Resolver(() => User)
export class UsersResolver {
  constructor(private readonly usersService: UsersService) {}

  @UseGuards(AuthGuard)
  @Query(() => User, { name: 'me' })
  me(@CurrentUser() user: User) {
    return this.usersService.findOne(user.id);
  }

  @UseGuards(AuthGuard)
  @Mutation(() => User)
  updateUser(
    @CurrentUser() user: User,
    @Args('updateUserInput')
    updateUserInput: UpdateUserInput
  ) {
    return this.usersService.update(id, updateUserInput);
  }
}
