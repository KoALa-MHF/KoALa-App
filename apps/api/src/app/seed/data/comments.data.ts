import { UsersData } from './users.data';

export const CommentsData = [
  {
    id: 1,
    owner: UsersData.sessionParticipant1,
    text: 'This is a comment from the Participant',
    annotationId: 1,
  },
  {
    id: 2,
    owner: UsersData.sessionOwner1,
    text: 'This is a comment from the Owner',
    annotationId: 1,
  },
];
