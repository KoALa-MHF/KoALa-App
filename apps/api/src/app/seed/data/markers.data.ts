import { MarkerType } from '../../markers/entities/marker.entity';
import { UsersData } from './users.data';

export const MarkersData = [
  {
    id: 1,
    type: MarkerType.EVENT,
    name: 'Marker 1',
    abbreviation: 'M1',
    description: 'Marker 1 Description',
    color: '#000000',
    icon: 'icon1',
    owner: UsersData.sessionOwner1,
    contentColor: '#FFFFFF',
  },
  {
    id: 2,
    type: MarkerType.RANGE,
    name: 'Marker 2',
    abbreviation: 'M2',
    description: 'Marker 2 Description',
    color: '#000000',
    icon: 'icon2',
    owner: UsersData.sessionOwner1,
    contentColor: '#FFFFFF',
  },
  {
    id: 3,
    type: MarkerType.SLIDER,
    name: 'Marker 3',
    abbreviation: 'M3',
    description: 'Marker 3 Description',
    color: '#000000',
    icon: 'icon3',
    rangeFrom: -5,
    rangeTo: 5,
    owner: UsersData.sessionOwner1,
    contentColor: '#FFFFFF',
  },
  {
    id: 4,
    type: MarkerType.EVENT,
    name: 'Marker 4',
    abbreviation: 'M4',
    description: 'Marker 4 Description',
    color: '#000000',
    icon: 'icon4',
    owner: UsersData.sessionOwner2,
    contentColor: '#FFFFFF',
  },
];
