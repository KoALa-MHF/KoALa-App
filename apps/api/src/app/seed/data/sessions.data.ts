import { SessionStatus } from '../../sessions/entities/session.entity';
import { UsersData } from './users.data';

export const SessionsData = [
  {
    id: 1,
    name: 'Session 1',
    owner: UsersData.sessionOwner1,
    status: SessionStatus.IN_PREPARATION,
  },
  {
    id: 2,
    name: 'Session 2',
    owner: UsersData.sessionOwner2,
    status: SessionStatus.CLOSED,
  },
  {
    id: 3,
    name: 'Session 3',
    owner: UsersData.sessionOwner1,
    status: SessionStatus.IN_PREPARATION,
    start: Date.now(),
    end: Date.now() + 1000000,
  },
  {
    id: 4,
    name: 'Session 4',
    owner: UsersData.sessionOwner1,
    status: SessionStatus.OPEN,
    start: Date.now() - 2000,
    end: Date.now() - 1000,
  },
  {
    id: 5,
    name: 'Session 5',
    owner: UsersData.sessionOwner1,
    status: SessionStatus.ARCHIVED,
  },
];
