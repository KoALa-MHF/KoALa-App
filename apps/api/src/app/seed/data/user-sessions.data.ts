import { UsersData } from './users.data';
import { SessionsData } from './sessions.data';

export const UserSessionsData = [
  {
    owner: UsersData.SESSION_PARTICIPANT_1,
    session: SessionsData[0],
  },
  {
    owner: UsersData.SESSION_PARTICIPANT_2,
    session: SessionsData[0],
  },
  {
    owner: UsersData.SESSION_OWNER_1,
    session: SessionsData[1],
  },
  {
    owner: UsersData.SESSION_PARTICIPANT_2,
    session: SessionsData[2],
  },
];
