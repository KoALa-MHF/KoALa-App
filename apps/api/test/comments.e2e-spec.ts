import request from 'supertest-graphql';
import gql from 'graphql-tag';
import { INestApplication } from '@nestjs/common';
import { setupTestApplication } from './mocks/test.util';
import { UsersData } from '../src/app/seed/data/users.data';

const QUERY_ANNOTATION_COMMENTS = gql`
  query Annotation($id: Int!) {
    annotation(id: $id) {
      id
      note
      userSession {
        id
        owner {
          id
          email
        }
      }
      comments {
        id
        text
        owner {
          id
          email
        }
      }
    }
  }
`;

const CREATE_COMMENT = gql`
  mutation CreateComment($createCommentInput: CreateCommentInput!) {
    createComment(createCommentInput: $createCommentInput) {
      text
    }
  }
`;

const UPDATE_COMMENT = gql`
  mutation UpdateComment($id: Int!, $updateCommentInput: UpdateCommentInput!) {
    updateComment(id: $id, updateCommentInput: $updateCommentInput) {
      text
    }
  }
`;

const REMOVE_COMMENT = gql`
  mutation RemoveComment($id: Int!) {
    removeComment(id: $id) {
      id
    }
  }
`;

const QUERY_ANNOTATION_COMMENTS_VARIABLES = {
  id: 1,
};

const CREATE_COMMENT_VARIABLES_SESSION_OWNER = {
  createCommentInput: {
    text: 'New comment from a session owner',
    annotationId: 1,
  },
};

const CREATE_COMMENT_VARIABLES_ANNOTATION_OWNER = {
  createCommentInput: {
    text: 'New comment from the annoation owner',
    annotationId: 1,
  },
};

const UPDATE_COMMENT_VARIABLES_SESSION_OWNER = {
  id: 2,
  updateCommentInput: {
    text: 'Updated comment session owner',
  },
};

const UPDATE_COMMENT_VARIABLES_ANNOTATION_OWNER = {
  id: 1,
  updateCommentInput: {
    text: 'Updated comment annotation owner',
  },
};

const REMOVE_COMMENT_VARIABLES_SESSION_OWNER = {
  id: 2,
};

const REMOVE_COMMENT_VARIABLES_ANNOTATION_OWNER = {
  id: 1,
};

describe('Comments (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    app = await setupTestApplication();
  });

  afterEach(async () => {
    await app.close();
  });

  describe('Query Annotation Comments', () => {
    it('Not authenticated user should get "Unauthorized" error', async () => {
      const { errors } = await request(app.getHttpServer())
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Unauthorized');
    });

    it('Session Owner user should get list of all comments', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES)
        .expectNoErrors();

      expect(data).toMatchSnapshot();
    });

    it('Annotation Owner user should get list of all comments', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES)
        .expectNoErrors();

      expect(data).toMatchSnapshot();
    });

    it('Authenticated user who is not a session owner or not the annotation owner should not get an answer', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner2.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });

    it('Authenticated user who is participant in the session but not the annotation owner should not get an answer', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant2.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });
  });

  describe('Create Comment', () => {
    it('Not authenticated user should get "Unauthorized" error', async () => {
      const { errors } = await request(app.getHttpServer())
        .mutate(CREATE_COMMENT)
        .variables(CREATE_COMMENT_VARIABLES_ANNOTATION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Unauthorized');
    });

    it('Session owner can create a new comment', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(CREATE_COMMENT)
        .variables(CREATE_COMMENT_VARIABLES_SESSION_OWNER)
        .expectNoErrors();

      const { data: dataQuery } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES)
        .expectNoErrors();

      expect(dataQuery).toMatchSnapshot();

      expect(data).toMatchSnapshot();
    });

    it('Annotation owner can create a new comment', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .mutate(CREATE_COMMENT)
        .variables(CREATE_COMMENT_VARIABLES_ANNOTATION_OWNER)
        .expectNoErrors();

      expect(data).toMatchSnapshot();

      const { data: dataQuery } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES)
        .expectNoErrors();

      expect(dataQuery).toMatchSnapshot();
    });

    it('Authenticated user who is participant in the session but not the annotation owner should not be allowed to create a comment', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant2.id}`, { type: 'bearer' })
        .mutate(CREATE_COMMENT)
        .variables(CREATE_COMMENT_VARIABLES_ANNOTATION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });
  });

  describe('Update Comment', () => {
    it('Not authenticated user should get "Unauthorized" error', async () => {
      const { errors } = await request(app.getHttpServer())
        .mutate(UPDATE_COMMENT)
        .variables(UPDATE_COMMENT_VARIABLES_SESSION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Unauthorized');
    });

    it('Session Owner can update its own comment', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(UPDATE_COMMENT)
        .variables(UPDATE_COMMENT_VARIABLES_SESSION_OWNER)
        .expectNoErrors();

      expect(data).toMatchSnapshot();
    });

    it('Annoation Owner can update its own comment', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .mutate(UPDATE_COMMENT)
        .variables(UPDATE_COMMENT_VARIABLES_ANNOTATION_OWNER)
        .expectNoErrors();

      expect(data).toMatchSnapshot();
    });

    it('Annoation Owner cannot update session owners comment', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .mutate(UPDATE_COMMENT)
        .variables(UPDATE_COMMENT_VARIABLES_SESSION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });

    it('Session Owner cannot update annotation owners comment', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(UPDATE_COMMENT)
        .variables(UPDATE_COMMENT_VARIABLES_ANNOTATION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });

    it('Authenticated user who is participant in the session but not the annotation owner should not be allowed to update a comment', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant2.id}`, { type: 'bearer' })
        .mutate(UPDATE_COMMENT)
        .variables(UPDATE_COMMENT_VARIABLES_ANNOTATION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });
  });

  describe('Remove Comment', () => {
    it('Not authenticated user should get "Unauthorized" error', async () => {
      const { errors } = await request(app.getHttpServer())
        .mutate(REMOVE_COMMENT)
        .variables(REMOVE_COMMENT_VARIABLES_ANNOTATION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Unauthorized');
    });

    it('Session Owner can remove own comment', async () => {
      const { data: sessionsBeforeDelete } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES)
        .expectNoErrors();
      expect(sessionsBeforeDelete).toMatchSnapshot();

      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(REMOVE_COMMENT)
        .variables(REMOVE_COMMENT_VARIABLES_SESSION_OWNER)
        .expectNoErrors();
      expect(data).toMatchSnapshot();

      const { data: sessionsAfterDelete } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES)
        .expectNoErrors();
      expect(sessionsAfterDelete).toMatchSnapshot();
    });

    it('Annotation Owner can remove own comment', async () => {
      const { data: sessionsBeforeDelete } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES)
        .expectNoErrors();
      expect(sessionsBeforeDelete).toMatchSnapshot();

      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .mutate(REMOVE_COMMENT)
        .variables(REMOVE_COMMENT_VARIABLES_ANNOTATION_OWNER)
        .expectNoErrors();
      expect(data).toMatchSnapshot();

      const { data: sessionsAfterDelete } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .query(QUERY_ANNOTATION_COMMENTS)
        .variables(QUERY_ANNOTATION_COMMENTS_VARIABLES)
        .expectNoErrors();
      expect(sessionsAfterDelete).toMatchSnapshot();
    });

    it('Session Owner cannot remove annotation owner comment', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(REMOVE_COMMENT)
        .variables(REMOVE_COMMENT_VARIABLES_ANNOTATION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });

    it('Annotation Owner cannot remove session owner comment', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .mutate(REMOVE_COMMENT)
        .variables(REMOVE_COMMENT_VARIABLES_SESSION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });

    it('Authenticated user who is participant in the session but not the annotation owner should not be allowed to remove a comment', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant2.id}`, { type: 'bearer' })
        .mutate(REMOVE_COMMENT)
        .variables(REMOVE_COMMENT_VARIABLES_ANNOTATION_OWNER);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden');
    });
  });
});
