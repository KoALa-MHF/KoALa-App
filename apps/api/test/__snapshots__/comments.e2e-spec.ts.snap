// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments (e2e) Create Comment Annotation owner can create a new comment 1`] = `
{
  "createComment": {
    "text": "New comment from the annoation owner",
  },
}
`;

exports[`Comments (e2e) Create Comment Annotation owner can create a new comment 2`] = `
{
  "annotation": {
    "comments": [
      {
        "id": 1,
        "owner": {
          "email": "session-participant-1@koala-app.de",
          "id": "3",
        },
        "text": "This is a comment from the Participant",
      },
      {
        "id": 2,
        "owner": {
          "email": "session-owner-1@koala-app.de",
          "id": "1",
        },
        "text": "This is a comment from the Owner",
      },
      {
        "id": 3,
        "owner": {
          "email": "session-participant-1@koala-app.de",
          "id": "3",
        },
        "text": "New comment from the annoation owner",
      },
    ],
    "id": 1,
    "note": "",
    "userSession": {
      "id": 6,
      "owner": {
        "email": "session-participant-1@koala-app.de",
        "id": "3",
      },
    },
  },
}
`;

exports[`Comments (e2e) Create Comment Session owner can create a new comment 1`] = `
{
  "annotation": {
    "comments": [
      {
        "id": 1,
        "owner": {
          "email": "session-participant-1@koala-app.de",
          "id": "3",
        },
        "text": "This is a comment from the Participant",
      },
      {
        "id": 2,
        "owner": {
          "email": "session-owner-1@koala-app.de",
          "id": "1",
        },
        "text": "This is a comment from the Owner",
      },
      {
        "id": 3,
        "owner": {
          "email": "session-owner-1@koala-app.de",
          "id": "1",
        },
        "text": "New comment from a session owner",
      },
    ],
    "id": 1,
    "note": "",
    "userSession": {
      "id": 6,
      "owner": {
        "email": "session-participant-1@koala-app.de",
        "id": "3",
      },
    },
  },
}
`;

exports[`Comments (e2e) Create Comment Session owner can create a new comment 2`] = `
{
  "createComment": {
    "text": "New comment from a session owner",
  },
}
`;

exports[`Comments (e2e) Query Annotation Comments Annotation Owner user should get list of all comments 1`] = `
{
  "annotation": {
    "comments": [
      {
        "id": 1,
        "owner": {
          "email": "session-participant-1@koala-app.de",
          "id": "3",
        },
        "text": "This is a comment from the Participant",
      },
      {
        "id": 2,
        "owner": {
          "email": "session-owner-1@koala-app.de",
          "id": "1",
        },
        "text": "This is a comment from the Owner",
      },
    ],
    "id": 1,
    "note": "",
    "userSession": {
      "id": 6,
      "owner": {
        "email": "session-participant-1@koala-app.de",
        "id": "3",
      },
    },
  },
}
`;

exports[`Comments (e2e) Query Annotation Comments Session Owner user should get list of all comments 1`] = `
{
  "annotation": {
    "comments": [
      {
        "id": 1,
        "owner": {
          "email": "session-participant-1@koala-app.de",
          "id": "3",
        },
        "text": "This is a comment from the Participant",
      },
      {
        "id": 2,
        "owner": {
          "email": "session-owner-1@koala-app.de",
          "id": "1",
        },
        "text": "This is a comment from the Owner",
      },
    ],
    "id": 1,
    "note": "",
    "userSession": {
      "id": 6,
      "owner": {
        "email": "session-participant-1@koala-app.de",
        "id": "3",
      },
    },
  },
}
`;

exports[`Comments (e2e) Remove Comment Annotation Owner can remove own comment 1`] = `
{
  "annotation": {
    "comments": [
      {
        "id": 1,
        "owner": {
          "email": "session-participant-1@koala-app.de",
          "id": "3",
        },
        "text": "This is a comment from the Participant",
      },
      {
        "id": 2,
        "owner": {
          "email": "session-owner-1@koala-app.de",
          "id": "1",
        },
        "text": "This is a comment from the Owner",
      },
    ],
    "id": 1,
    "note": "",
    "userSession": {
      "id": 6,
      "owner": {
        "email": "session-participant-1@koala-app.de",
        "id": "3",
      },
    },
  },
}
`;

exports[`Comments (e2e) Remove Comment Annotation Owner can remove own comment 2`] = `
{
  "removeComment": {
    "id": 1,
  },
}
`;

exports[`Comments (e2e) Remove Comment Annotation Owner can remove own comment 3`] = `
{
  "annotation": {
    "comments": [
      {
        "id": 2,
        "owner": {
          "email": "session-owner-1@koala-app.de",
          "id": "1",
        },
        "text": "This is a comment from the Owner",
      },
    ],
    "id": 1,
    "note": "",
    "userSession": {
      "id": 6,
      "owner": {
        "email": "session-participant-1@koala-app.de",
        "id": "3",
      },
    },
  },
}
`;

exports[`Comments (e2e) Remove Comment Session Owner can remove own comment 1`] = `
{
  "annotation": {
    "comments": [
      {
        "id": 1,
        "owner": {
          "email": "session-participant-1@koala-app.de",
          "id": "3",
        },
        "text": "This is a comment from the Participant",
      },
      {
        "id": 2,
        "owner": {
          "email": "session-owner-1@koala-app.de",
          "id": "1",
        },
        "text": "This is a comment from the Owner",
      },
    ],
    "id": 1,
    "note": "",
    "userSession": {
      "id": 6,
      "owner": {
        "email": "session-participant-1@koala-app.de",
        "id": "3",
      },
    },
  },
}
`;

exports[`Comments (e2e) Remove Comment Session Owner can remove own comment 2`] = `
{
  "removeComment": {
    "id": 2,
  },
}
`;

exports[`Comments (e2e) Remove Comment Session Owner can remove own comment 3`] = `
{
  "annotation": {
    "comments": [
      {
        "id": 1,
        "owner": {
          "email": "session-participant-1@koala-app.de",
          "id": "3",
        },
        "text": "This is a comment from the Participant",
      },
    ],
    "id": 1,
    "note": "",
    "userSession": {
      "id": 6,
      "owner": {
        "email": "session-participant-1@koala-app.de",
        "id": "3",
      },
    },
  },
}
`;

exports[`Comments (e2e) Update Comment Annoation Owner can update its own comment 1`] = `
{
  "updateComment": {
    "text": "Updated comment annotation owner",
  },
}
`;

exports[`Comments (e2e) Update Comment Session Owner can update its own comment 1`] = `
{
  "updateComment": {
    "text": "Updated comment session owner",
  },
}
`;
